dyno-core/src/main/java/com/netflix/dyno/connectionpool/impl/health/SimpleErrorMonitorImpl.java

```diff
@@ -17,47 +17,56 @@
 
 import java.util.concurrent.atomic.AtomicInteger;
 
-public class SimpleErrorMonitorImpl implements ErrorMonitor {
+	public class SimpleErrorMonitorImpl implements ErrorMonitor {
 
-	private final AtomicInteger errorCount = new AtomicInteger(0);
-	private final int threshold;
-	
-	public SimpleErrorMonitorImpl(int numErrorThreshold) {
-		threshold = numErrorThreshold;
-	}
-	
-	@Override
-	public boolean trackError(int numErrors) {
-		
-		int currentCount = errorCount.addAndGet(numErrors);
-		if (currentCount >= threshold) {
-			// Reset the count
-			boolean success = errorCount.compareAndSet(currentCount, 0);
-			if (success) {
-				return false;  // ERROR above threshold! 
-			} else {
-				return true;   // all OK. Someone else beat us to reporting the errors as above threshold
+		private final AtomicInteger errorCount = new AtomicInteger(0);
+		private final int threshold;
+		private final int maxValue;
```

Comment:
Could you add some comments to give more context here? What's the difference between maxValue and threshold?

Reply:
refactored this code to use maxValue instead of threshold